### PR TITLE
Add support for access token and cleanup other configs

### PR DESCRIPTION
**Pull request by pellared**
## Why

The user may want to send traces directly to Splunk back-end.

## What

- Add support for the [access token](https://docs.splunk.com/observability/admin/authentication-tokens/org-tokens.html) so that allows sending traces directly to Splunk back-end.
- Refine other configs to improve the compliance with [GDI specification](https://github.com/signalfx/gdi-specification/blob/main/specification/configuration.md)

## Testing

Added integration tests using `httptest.Server`.

Tested manually with real Jaeger and Splunk-backend with @owais (thank you!). Results below: 

image

image

## Remarks

@MrAlias 

Bug in OTel Go SDK discovered. Basically it is defaulting to the gRPC endpoint http://localhost:14250 instead of the HTTP thrift endpoint http://localhost:14268/api/traces .

Here is a PR in OTel Go repository: https://github.com/open-telemetry/opentelemetry-go/pull/1898

## Next PR

- Add an example that can be used e.g. during the demo

### CODE REVIEW

**Comment by pellared** [comment]
@MrAlias can you review it, please?

**Review by owais — APPROVED** [review]

**Comment by iNikem** [comment]
@owais @MrAlias anything preventing from merging this?

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,12 +8,31 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## [Unreleased]
 
+### Added
+
+- Add support for setting the [Splunk's organization access token](https://docs.splunk.com/observability/admin/authentication-tokens/org-tokens.html)
+  using the `SPLUNK_ACCESS_TOKEN` environmental variable or `contrib.WithEndpoint` option.
+  It allows exporters sending data directly to the Splunk back-end.
+  To do so, the `OTEL_EXPORTER_JAEGER_ENDPOINT` or `contrib.WithEndpoint` must be set
```
Comment (MrAlias) [inline CHANGELOG.md:16]:
```suggestion
  using the `SPLUNK_ACCESS_TOKEN` environmental variable or `distro.WithAccessToken` option.
  It allows exporters sending data directly to the Splunk back-end.
  To do so, the `OTEL_EXPORTER_JAEGER_ENDPOINT` or `distro.WithEndpoint` must be set
```

right?

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,12 +8,31 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
+  with Splunk back-end ingest endpoint URL: `https://ingest.<REALM>.signalfx.com/v2/trace`.
+
+### Changed
+
+- The default Jaeger exporter URL is now loaded by
+  [`go.opentelemetry.io/otel/exporters/trace/jaeger`](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/trace/jaeger)
+  module.
+- Applying `contrib.WithEndpoint("")` results in a no operation.
```
Comment (MrAlias) [inline CHANGELOG.md:24]:
```suggestion
- Applying `distro.WithEndpoint("")` results in no operation.
```

File: README.md [hunk 3]
```diff
@@ -58,6 +58,17 @@ func main() {
 Documentation on how to manually instrument a Go application is available
 [here](https://opentelemetry.io/docs/go/getting-started/).
 
+## Splunk specfic configuration
```
Comment (MrAlias) [inline README.md:61]:
```suggestion
## Splunk specific configuration
```

File: README.md [hunk 4]
```diff
@@ -58,6 +58,17 @@ func main() {
...
+
+| Environment variable      | Option             | Default value  | Description |
+| ------------------------- | -------------------| -------------- | ---------------------------------------------------------------------- |
+| `SPLUNK_ACCESS_TOKEN`     | `WithAccessToken`  |                | The [Splunk's organization access token](https://docs.splunk.com/observability/admin/authentication-tokens/org-tokens.html). [[1](#cfg1)] |
+
+[<a name="cfg1">1</a>]: The [Splunk's organization access token](https://docs.splunk.com/observability/admin/authentication-tokens/org-tokens.html)
+allows exporters sending data directly to the [Splunk Observability Cloud](https://dev.splunk.com/observability/docs/apibasics/api_list/).
+To do so, the `OTEL_EXPORTER_JAEGER_ENDPOINT` or `contrib.WithEndpoint` must be set
+with Splunk back-end ingest endpoint URL: `https://ingest.<REALM>.signalfx.com/v2/trace`.
```
Comment (MrAlias) [inline README.md:70]: Are we planning to add support of `SPLUNK_REALM` in a subsequent PR?

File: distro/config.go [hunk 5]
```diff
@@ -84,19 +73,30 @@ func envOr(key, alt string) string {
...
+// Setting empty string results in no operation.
 func WithEndpoint(endpoint string) Option {
-	return func(c *config) {
+	return optionFunc(func(c *config) {
 		c.Endpoint = endpoint
-	}
+	})
+}
+
+// WithAccessToken configures the auth token
```
Comment (MrAlias) [inline distro/config.go:95]:
```suggestion
// WithAccessToken configures the authentication token
```

File: CHANGELOG.md (hunk 1 above)
Comment (pellared) [reply CHANGELOG.md:16]: yup 👍

File: README.md (hunk 4 above)
Comment (pellared) [reply README.md:70]: AFAIK it is required only for Data Collector. Not for instrumentation libraries. Reference: https://github.com/signalfx/gdi-specification/blob/main/specification/configuration.md

Comment (MrAlias) [reply README.md:70]: True. Just curious your thoughts.

**Review by MrAlias — APPROVED** [review]

**Pull request merged**